csupport/addC.go

```diff
@@ -0,0 +1,101 @@
+package csupport
+
+import (
+	"canaveral/lib"
+	"fmt"
+	"io/ioutil"
+	"os"
+)
+
+// createMainFile creates a basic projName.c file that uses stdlib.h
+// to print "Canaveral is awesome!" in the main function
+// * tested
+func createMainFile(projName string) error {
+	err := lib.CreateFile(projName + ".c")
+	if err != nil {
+		return err
+	}
+	mainfile, err := os.OpenFile(projName+".c", os.O_APPEND|os.O_WRONLY, 0644)
+	defer func() {
+		err := mainfile.Close()
+		lib.Check(err)
+	}()
+
+	include := "#include <stdio.h>"
+	main := "int main() {\n\tprintf(\"Canaveral is awesome!" + `\n` + "\");\n\treturn 0;\n}\n"
+
+	_, err = mainfile.WriteString(include + "\n\n" + main)
+	return err
+}
+
+// createREADME creates a README.md file
+// * tested
+func createREADME(projName string) error {
+	err := lib.CreateFile("README.md")
+	if err != nil {
+		return err
+	}
+	readme, err := os.OpenFile("README.md", os.O_APPEND|os.O_WRONLY, 0644)
+	defer func() {
+		err := readme.Close()
+		lib.Check(err)
+	}()
+	header := "# " + projName
+	making := "## Compiling\nTo compile " + projName + "simply download it and run `make`. To clean old builds, run `make clean`"
+	canaveral := "###### Created with [Canaveral](https://github.com/jchengjr77/canaveral)"
+	_, err = readme.WriteString(header + "\n\n" + making + "\n\n" + canaveral)
+	return err
+}
+
+// createMakeFile creates a make file that compiles the projName.c file into
+// a binary called projName
+// * tested
+func createMakeFile(projName string) error {
```

Comment:
AND a Makefile! A true goat.

Reply:
It took me so long to understand how all of the makefile features worked hahaha. Thoughts on adding a simple function like `canaveral addmake [filename].c` that adds [filename] to the compilation script in the makefile? It would probably only be able to handle very basic stuff so it would only be very useful for really small projects, but might make it easier to have medium sized projects without having to worry too much about knowing or creating a makefile. I don't know for sure how much work it would be though, so maybe we can discuss it on our call.

Reply:
Will discuss during our call